### PR TITLE
conmon: do not put NULLs in log lines

### PR DESCRIPTION
**Pull request by runcom**
For kubernetes-incubator/cri-o#430

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

File: conmon/conmon.c [hunk 1]
```diff
@@ -108,10 +108,17 @@ int set_k8s_timestamp(char *buf, ssize_t buflen, const char *stream_type)
...
 		off = -off;
 	}
-	snprintf(buf, buflen, "%d-%02d-%02dT%02d:%02d:%02d%c%02d:%02d %s ",
+	snprintf(buf, buflen, "%d-%02d-%02dT%02d:%02d:%02d%c%02d:%02d %s %s",
 		tm->tm_year + 1900, tm->tm_mon + 1, tm->tm_mday,
 		tm->tm_hour, tm->tm_min, tm->tm_sec,
-		off_sign, off / 3600, off % 3600, stream_type);
+		off_sign, off / 3600, off % 3600, stream_type, log);
+
+	char newbuf[buflen-1];
```
Comment (mrunalp) [inline conmon/conmon.c]: I don't think we need to allocate a new buffer here.

File: conmon/conmon.c [hunk 2]
```diff
@@ -108,10 +108,17 @@ int set_k8s_timestamp(char *buf, ssize_t buflen, const char *stream_type)
...
+	strncpy(newbuf, buf, buflen-1);
+
+	if (write(logfd, newbuf, buflen-1) != buflen-1) {
```
Comment (mrunalp) [inline conmon/conmon.c]: This should work with just `buf`, right?

Comment (runcom) [reply conmon/conmon.c]: Right

**Comment by runcom** [comment]
Just in case, I've fixed the comments here.

**Comment by mrunalp** [comment]
There are issues with this one:
```
2017-04-07T09:52:34-07:00 stdout KUBERNETES_SERVICE_PORT=443
2017-04-07T09:52:34-07:00 stdout KUBERNETES_PORT=tcp://10.0.0.1:443
2017-04-07T09:52:34-07:00 stdout FOOBAR=foo-value;;bar-value
2017-04-07T09:52:34-07:00 stdout SHLVL=1
2017-04-07T09:52:34-07:00 stdout HOME=/root
2017-04-07T09:52:34-07:00 stdout TERM=xterm
2017-04-07T09:52:34-07:00 stdout KUBERNETES_PORT_443_TCP_ADDR=10.0.0.1
2017-04-07T09:52:34-07:00 stdout BAR=bar-value
2017-04-07T09:52:34-07:00 stdout PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2017-04-07T09:52:34-07:00 stdout KUBERNETES_PORT_4/tmp/conmon-term.XXXXXXXX
43_TCP_PORT=443
2017-04-07T09:52:34-07:00 stdout FOO=foo-value
2017-04-07T09:52:34-07:00 stdout KUBERNETES_PORT_443_TCP_PROTO=tcp
2017-04-07T09:52:34-07:00 stdout KUBERNETES_PORT_443_TCP=tcp://10.0.0.1:443
2017-04-07T09:52:34-07:00 stdout KUBERNETES_SERVICE_PORT_HTTPS=443
2017-04-07T09:52:34-07:00 stdout PWD=/
2017-04-07T09:52:34-07:00 stdout KUBERNETES_SERVICE_HOST=10.0.0.1
2017-04-07T09:52:34-07:00 stdout sr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2017-04-07T09:52:34-07:00 stdout KUBERNETES_PORT_4/tmp/conmon-term.XXXXXXXX
```

**Comment by mrunalp** [comment]
We do need the \0 at the end of the buf or old contents will be printed. I'll fix it up in my cri-o PR.

**Pull request closed without merging**